src/rez/package_search.py

```diff
@@ -152,7 +153,7 @@ def get_plugins(package_name, paths=None):
             continue  # not a plugin of itself
 
         plugin_pkg = get_latest_package(package_name_, paths=paths)
-        if not plugin_pkg.plugin_for:
+        if not plugin_pkg or not plugin_pkg.plugin_for:
```

Comment:
I'm wondering if we should instead make sure that `get_latest_package` returns valid packages. The docstring of `get_latest_package` says that it should either return a `Package` object or None is a package is not found. There is also this TODO https://github.com/AcademySoftwareFoundation/rez/blob/71d990bc63cd9314114e45c2002fa15656a778b0/src/rez/packages.py#L925-L926

It my books, that's what I call smell.

Reply:
In the case of #1618 I would say that get_latest_package is correctly returning None, as there is no valid package in an empty folder. Perhaps then iter_package_families should be more thorough and not return package names if there is no valid package in the folders?